src/sql/workbench/parts/query/browser/gridPanel.ts

```diff
@@ -419,7 +419,23 @@ export abstract class GridTableBase<T> extends Disposable implements IView {
 		this.scrolled = false;
 	}
 
-	private build(): void {
+	// actionsOrientation controls the orientation (horizontal or vertical) of the actionBar
+	private build(actionsOrientation?: ActionsOrientation): void {
+
+		// Default is VERTICAL
+		if (isUndefinedOrNull(actionsOrientation)) {
+			actionsOrientation = ActionsOrientation.VERTICAL;
+		}
+
+		let actionBarContainer = document.createElement('div');
+
+		// Create a horizontal actionbar if orientation passed in is HORIZONTAL
+		if (actionsOrientation === ActionsOrientation.HORIZONTAL) {
```

Comment:
Could you do the same thing by just adding css rules and applying a class like `vertical` or `horizontal` to the bar? Generally if the properties aren't calculated, I prefer css.

Reply:
sure thing.